File: src/main/java/io/protobase/f7/formulas/info/NA.java
package io.protobase.f7.formulas.info;

import io.protobase.f7.errors.NAException;
import io.protobase.f7.formulas.AbstractFormula;
import io.protobase.f7.formulas.FormulaName;
import io.protobase.f7.models.GridColumnRowKey;

import java.util.function.BiFunction;
import java.util.function.Function;

public class NA extends AbstractFormula {
  public static FormulaName NAME = FormulaName.NA;
  public static NA SELF = new NA();

  public NA() {
    super();
  }

  public NA(Function<Object, Object> lookup, BiFunction<GridColumnRowKey, Object, Object> collateralLookup) {
    super(lookup, collateralLookup);
  }

  @Override
  public Object internal(GridColumnRowKey origin, Object... args) {
    checkLength(args.length, 0, NAME);
    return new NAException();
  }
}

File: src/main/js/formulas/info/NA.ts
import { NAException } from "../../errors/NAException";
import { SheetColumnRowKey } from "../../models/common/SheetColumnRowKey";
import { Complex } from "../../models/common/Types";
import { AbstractFormula } from "../AbstractFormula";
import { FormulaName } from "../FormulaName";

export class NA extends AbstractFormula {
  static SELF: NA = new NA();
  NAME = FormulaName.NA;

  internal(origin: SheetColumnRowKey, ...values: Array<Complex>) {
    AbstractFormula.checkLength(values.length, 0, this.NAME);
    return new NAException();
  }
}
